src/Controller/EntityAutocompleteController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Drupal\select2\Controller;
+
+use Drupal\Component\Utility\Crypt;
+use Drupal\Component\Utility\Tags;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Site\Settings;
+use Drupal\select2\EntityAutocompleteMatcher;
+use Drupal\Core\KeyValueStore\KeyValueStoreInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Defines a route controller for entity autocomplete form elements.
+ */
+class EntityAutocompleteController extends ControllerBase {
```

Comment:
Is it possible to extend of original EntityAutocompleteController? Overriding ``::create`` should be enough to load our own matcher.

Reply:
I thought that in the first place as well. But we have to expose a results key in the matches array. So we have to overwrite that method :(